updated relation docs according to PORT-1765 (#123)
port-labs/port-docs: docs/software-catalog/relation/relation.md

| `required` | `Boolean` | Boolean flag to define whether the target Entity is a must when creating a new Entity from the source Blueprint                                                |
| `many`     | `Boolean` | Boolean flag to define whether multiple target Entities can be mapped to the Relation. For more information refer to [X-to-many relation](#x-to-many-relation) |

:::note

> A relation with `many` field set to `true` cannot be required.

Reviewer: ```suggestion
A Relation can't be configured with both `many` and `required` set to `true`
```
